lib/gutentag/active_record/instance_methods.rb

```diff
@@ -9,7 +9,8 @@ module Gutentag::ActiveRecord::InstanceMethods
   # the instance directly (e.g. article.tags << tag), which invokes the save
   # callbacks, but the old tag_names value is stored but not updated.
   def reset_tag_names
-    return if tag_names_previously_changed?
+    return if ActiveRecord::VERSION::STRING.to_f < 5.1 && tag_names_previously_changed?
+    return if ActiveRecord::VERSION::STRING.to_f >= 5.1 && saved_change_to_tag_names?
```

Comment:
What do you think of this solution instead?

Reply:
Looks pretty solid functionally - one tweak from a syntax perspective: shift `rails_version` to be a constant (so it's only evaluated once, rather than many times). And perhaps the name can be slightly more specific: `AR_VERSION` or `ACTIVE_RECORD_VERSION`?

Or, as I think about it in more depth… maybe we instead just add in a method alias if an older Rails version is in play? Removes the need for the extra constant too…

```ruby
if ActiveRecord::VERSION::STRING.to_f < 5.1
  alias :tag_names_previously_changed? :saved_change_to_tag_names?
end
```

This means that the call within `reset_tag_names` can refer to the most recent name (`saved_change_to_tag_names?`), and the alias eventually gets removed completely if/when Gutentag drops Rails 5.1 support.